### PR TITLE
Update docs for enabling sphinx.ext.autosummary

### PR DESCRIPTION
**Pull request by yunchu**
### Summary

enable sphinx autosummary extension to generate API reference pages.

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

File: src/otx/algorithms/classification/tools/__init__.py
Comment (goodsong81) [inline src/otx/algorithms/classification/tools/__init__.py:1]: Intentionally removed? It's quite handy when debugging the installed package.

**Review by goodsong81 — COMMENTED** [review]
Looks quite nice automation, indeed!
By the way, I have a minor question.

File: src/otx/algorithms/classification/tools/__init__.py
Comment (yunchu) [reply src/otx/algorithms/classification/tools/__init__.py:1]: actually, that's my concerns. it leads some warning or error while generating docs. I will try again without removing some init files. thanks for checking!

**Review by goodsong81 — DISMISSED** [review]

File: tests/unit/algorithms/detection/test_helpers.py [hunk 1]
```diff
@@ -33,7 +33,7 @@
     "src/otx/algorithms/detection/configs/instance_segmentation", "efficientnetb2b_maskrcnn"
 )
 DEFAULT_DET_RECIPE_CONFIG_PATH = "src/otx/recipes/stages/detection/incremental.py"
-DEFAULT_ISEG_RECIPE_CONFIG_PATH = "src/otx/recipes/stages/instance-segmentation/incremental.py"
+DEFAULT_ISEG_RECIPE_CONFIG_PATH = "src/otx/recipes/stages/instance_segmentation/incremental.py"
```
Comment (goodsong81) [inline tests/unit/algorithms/detection/test_helpers.py:36]: These path changes are due to documentation warnings? Interesting.

**Review by goodsong81 — APPROVED** [review]

File: tests/unit/algorithms/detection/test_helpers.py (hunk 1 above)
Comment (yunchu) [reply tests/unit/algorithms/detection/test_helpers.py:36]: yes. sphinx tried to find `otx.recipes.stages.instance` module when the path name includes `-`.

**Pull request merged**